chore: Cleanup Readme. Add Sponsor section. (#8269)
tauri-apps/tauri: README.md

[![https://good-labs.github.io/greater-good-affirmation/assets/images/badge.svg](https://good-labs.github.io/greater-good-affirmation/assets/images/badge.svg)](https://good-labs.github.io/greater-good-affirmation)
[![support](https://img.shields.io/badge/sponsor-Open%20Collective-blue.svg)](https://opencollective.com/tauri)

## Introduction

Tauri is a framework for building tiny, blazingly fast binaries for all major desktop platforms. Developers can integrate any front-end framework that compiles to HTML, JS and CSS for building their user interface. The backend of the application is a rust-sourced binary with an API that the front-end can interact with.

> The user interface in Tauri apps currently leverages [`tao`](https://docs.rs/tao) as a window handling library on macOS and Windows, and [`gtk`](https://gtk-rs.org/docs/gtk/) on Linux via the **Tauri-team** incubated and maintained [WRY](https://github.com/tauri-apps/wry), which creates a unified interface to the system webview (and other goodies like Menu and Taskbar), leveraging WKWebView on macOS & iOS, WebView2 on Windows, WebKitGTK on Linux and Android System WebView on Android.

Reviewer: ```suggestion
The user interface in Tauri apps currently leverages [`tao`](https://docs.rs/tao) as a window handling library on macOS, Windows, Linux, Android and iOS. To render your application, Tauri uses [WRY](https://github.com/tauri-apps/wry), a library which provides a unified interface to the system webview, leveraging WKWebView on macOS & iOS, WebView2 on Windows, WebKitGTK on Linux and Android System WebView on Android.
```

To learn more about the details of how all of these pieces fit together, please consult this [ARCHITECTURE.md](https://github.com/tauri-apps/tauri/blob/dev/ARCHITECTURE.md) document.

## Features

The list of Tauri's features includes, but is not limited to:

> - Built-in app bundler to create app bundles in formats like `.app`, `.deb`, and `.msi`.

Reviewer: ```suggestion
- Built-in app bundler to create app bundles in formats like `.app`, `.dmg`, `.deb`, `.AppImage` and Windows installers like `.exe` (via NSIS) and `.msi` (via WiX).
```

Reviewer: i guess you removed AppImage intentionally :joy:

Author: i intentionally didn't list all formats and changed the wording to "like ..." to make it clearer that it's only a subset. I figured the list would get quite long if we add .pkg .rpm .snap flatpak/flathub and maybe mobile formats.

Author: but ig if we leave the wording as is it's still fine to re-add the v1 formats and not add the upcoming formats (we'll forget anyway)

Reviewer: cool it's ok the way it is

| :----------------------- | :-------------- |
| Windows                  | 7 and above     |
| macOS                    | 10.15 and above |

> | Linux                    | webkit2gtk 4.0 for Tauri v1 (for example Ubuntu 18.04). webkit2gtk 4.1 for Tauri v2 (for example Ubuntu 22.04). |

Reviewer 2: It does seem like we're losing some nuance here.
Perhaps this table (for the other platforms as well) could be linked to a docs page?

That said I agree with summarizing, maybe even removing it from the README if we do move it to a page.
My thought is for a developer that *wants* to know the nuances, how would they find out?

Author: > My thought is for a developer that wants to know the nuances, how would they find out?

ig they have to go to one of those cross-distro package list websites and search for webkitgtk versions 🙃 Same as before really. The runtime package names imho can be easily derived from the dev packages listed in the prereqs docs

i don't think the version before really had much more value (especially since it still mentioned ubuntu1804 for tauri v2 which is incorrect).
It's not like we actually support void, none of us know anyone even running it i think?
Listing Arch and fedora also doesn't hold much value anymore, only installations that weren't updated for over a year now don't have webkitgtk 4.1. However they are more supported than void, Lucas runs arch, i run fedora (and ubuntu), and ubuntu is used in wsl and ci etc.

So my problem here is that i don't know _what_ to tell users, and _how_, at least as long as the actual webkitgtk version doesn't really matter (meaning the 2.38 like version) in tauri.

We could also remove the mention of webkitgtk 4.0 since technically this is the v2 branch.

Author: I am however also unhappy with the change so i'm open for ideas

Reviewer 3: Would this be a place where we keep a single source of truth on the website (the prerequisites page perhaps) and link to that for more information? That also let's you be a bit more verbose on any difference of requirements for _running_ vs. _developing_ a Tauri app.

| iOS/iPadOS (alpha)       | 9 and above     |
| Android (alpha)          | 7 and above     |

## Getting Started

If you are interested in making a tauri app, please visit the [documentation website](https://tauri.app).

The quickest way to get started is to install the [prerequisites](https://tauri.app/v1/guides/getting-started/prerequisites) for your system and create a new project with [`create-tauri-app`](https://github.com/tauri-apps/create-tauri-app/#usage). For example with `npm`:

```sh
npm create tauri-app@latest

> ```

Reviewer 3: I would move this to be up as close to the top as possible to allow someone to immediately jump right in.

### Documentation

> Documentation in a polyglot system is a tricky proposition. To this end, we prefer to use inline documentation of Rust code and JSDoc in TypeScript / JavaScript code. We autocollect these and publish them using [starlight](https://starlight.astro.build/) and [netlify](https://netlify.com). Here is the hosting repository for the documentation site: https://github.com/tauri-apps/tauri-docs

Reviewer 3: I don't believe we use JSDoc, just Typedoc. But could just remove the "JSDoc in" section.

I'd also remove the `We autocollect these and publish them using [starlight](https://starlight.astro.build/) and [netlify](https://netlify.com).` line and just keep the one to point to the docs repo where that respective info is since it's not really relevant to this repo.

Author: does this work? (stumbling over the wording...)

Reviewer 3: Perfect, thanks!

## Partners

<table>
  <tbody>
    <tr>
      <td align="center" valign="middle">
        <a href="https://crabnebula.dev" target="_blank">
          <img src=".github/sponsors/crabnebula.svg" alt="CrabNebula" width="283">
        </a>
      </td>
    </tr>
  </tbody>
</table>

> For the complete list of sponsors please visit our [website](https://tauri.app#sponsors) and [Open Collective](https://opencollective.com/tauri).

Reviewer 4: There's also sponsors to be found on Github: https://github.com/sponsors/tauri-apps
Maybe we should list that as well.

Open Collective only shows the monthly payouts so you'd have to look on Github to see a breakdown of who the sponsors are.

Author: That's true, i tried to get it in at first but my english failed me on that. Maybe we can cover that with Lorenzo's sponsor review at which point we will probably remove the opencollective mention anyway. Also, github sponsors are visible on the repo so it's that's something.

Anyway, if you can come with a good wording that doesn't sound like i wrote it i'm happy to add it as well.

> For a list of more of our sponsors please visit our [website](https://tauri.app#sponsors) and [Open Collective](https://opencollective.com/tauri).

Reviewer: ```suggestion
For the complete list of sponsors please visit our [website](https://tauri.app#sponsors) and [Open Collective](https://opencollective.com/tauri).
```
